#**1. Import libaries and Load data**

**📥 Import libaries**

In [1]:
import pandas as pd

🔄 Load data

The datasets are in the shared folder on Google Drive:

https://drive.google.com/drive/folders/1qhOW9Y2orRXuzbX-kXEmuJ7TMQiRs2Uv

- **Table 1:** Monthly expenses

https://docs.google.com/spreadsheets/d/10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw/edit?gid=0#gid=0

In [2]:
google_sheet_id = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
monthly_expenses = pd.read_excel(url, sheet_name='Sheet1')

- **Table 2:** Payroll

https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/edit?gid=0#gid=0

In [3]:
google_sheet_id = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
payroll = pd.read_excel(url, sheet_name='Sheet1')

- **Table 3:** daily_marketing_spendings

https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/edit?gid=1513891606#gid=1513891606

In [4]:
google_sheet_id = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
daily_marketing = pd.read_excel(url, sheet_name='Sheet1')

- **Table 4:** receipts_history

https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/edit?gid=952917703#gid=952917703

In [5]:
google_sheet_id = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
payments = pd.read_excel(url, sheet_name='Sheet1')

- **Table 5:** customer_lifespan_data

https://docs.google.com/spreadsheets/d/1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI/edit?gid=1166661477#gid=1166661477

In [6]:
google_sheet_id = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
lifespan = pd.read_excel(url, sheet_name='Sheet1')

#**2. Metric Calculate**

##**2.1. Customer Acquisition Cost (CAC)**

**Customer Acquisition Cost (CAC)**: The cost associated with acquiring a new customer, including marketing and sales expenses.
- **CAC = [Total Sales and Marketing Expenses] / [Number Of New Customers Acquired]**

### 2.1.1. Total Sales and Marketing Expenses

Using these metrics to better understand where sales and marketing budget is going — tools, people, and daily operations. It helps identify which area drives results and where to optimize.

#### a. Marketing Software Expense

Check 'Monthly expenses' dataset:

In [7]:
monthly_expenses.sample(5)

#      month           category             item  amount
22  23 2023-03-01  Software Licenses       Salesforce    1700
2    3 2023-01-01  Software Licenses   Atlassian Jira    1200
20  21 2023-03-01  Software Licenses   Atlassian Jira    1400
8    9 2023-01-01              Other  Travel Expenses    3000
15  16 2023-02-01      Office Rental      Office Rent   10000

👉 Salesforce is classified as a marketing cost because it is used for customer relationship management and running marketing campaigns.

In [8]:
# Filter expense in March
expense_202303 = monthly_expenses[monthly_expenses['month'] == '2023-03-01']

In [9]:
# Find total 'Salesforce' expense in March
crm_expense_202303 = expense_202303[expense_202303['item'] == 'Salesforce'] ['amount'].sum()

# Check result
crm_expense_202303

np.int64(1700)

#### b. Sales & Marketing Salaries

Check 'Payroll' dataset:

In [10]:
payroll.sample(5)

month   department employee_name          position  paid
2  2023-01-01        Sales     Jim Brown   Sales Associate   700
34 2023-03-01        Sales      John Doe     Sales Manager  1500
17 2023-02-01        Sales      John Doe     Sales Manager  1500
8  2023-01-01  Engineering     Eva White  Junior Developer  1000
3  2023-01-01        Sales  Laura Miller   Sales Associate   800

In [11]:
# Filter payroll in March
payroll_202303 = payroll[payroll['month'] == '2023-03-01']

In [12]:
# Find total payroll of 'Sales' and 'Marketing' department in March
sales_marketing_salary_202303 = payroll_202303[(payroll_202303['department'] == 'Sales') |\
                                               (payroll_202303['department'] == 'Marketing')] ['paid'].sum()

# Check result
sales_marketing_salary_202303

np.int64(5950)

#### c. Daily Marketing Expense

Check 'daily_marketing_spendings' dataset:

In [13]:
daily_marketing.sample(5)

date       channel  spending
141 2023-02-05  Facebook Ads       839
104 2023-01-27    Google Ads       819
334 2023-03-25  LinkedIn Ads       373
213 2023-02-23  Facebook Ads       695
170 2023-02-12  LinkedIn Ads       689

In [14]:
# Find total daily marketing spending in March
daily_marketing_202303 = daily_marketing[(daily_marketing['date'] >= '2023-03-01') &\
                                         (daily_marketing['date'] <= '2023-03-31')] ['spending'].sum()

# Check result
daily_marketing_202303

np.int64(68830)

### 2.1.2. Number of New Customers

Check 'receipts_history' dataset:

In [15]:
payments.sample(5)

date  customer_id  receipt_amount  new_customer
67  2023-01-08         2733             209             0
92  2023-01-11         2365             111             0
866 2023-03-25         2798             418             0
630 2023-03-02         1282             471             0
683 2023-03-06         1657             175             0

In [16]:
# Filter receipts in March
payments_202303 = payments[(payments['date'] >= '2023-03-01') &\
                           (payments['date'] <= '2023-03-31')]

In [17]:
# Find 'new customers' in March
number_of_new_customers_202303 = payments_202303[payments_202303['new_customer'] == 1] ['customer_id'].nunique()

# Check result
number_of_new_customers_202303

63

### 2.1.3. CAC Calculation

In [18]:
# Calculate total sales and marketing expense
total_sales_and_marketing_cost = crm_expense_202303 + sales_marketing_salary_202303 + daily_marketing_202303

# Calculate CAC
cac = round(total_sales_and_marketing_cost / number_of_new_customers_202303, 2)

# Check result
cac

np.float64(1213.97)

✔️ The CAC of 1213.97 in March means that the company spent an average of 1213.97 to acquire each new customer during that month.

##**2.2. Average Revenue Per User (ARPU)**

**Average Revenue Per User (ARPU)**: The average revenue generated per user or customer over a specific period (e.g., monthly, quarterly).
- **ARPU = [Total Revenue] / [Number of Users]**

### 2.2.1. Total Revenue

In [19]:
# Calculate total revenue in March
total_revenue_202303 = payments_202303['receipt_amount'].sum()

# Check result
total_revenue_202303

np.int64(83033)

### 2.2.2. Number of Customers

In [20]:
# Calculate number of customers in March
number_of_customers_202303 = payments_202303['customer_id'].nunique()

# Check result
number_of_customers_202303

292

### 2.2.3. ARPU Calculation

In [21]:
# Calculate ARPU
arpu = round(total_revenue_202303 / number_of_customers_202303, 2)

# Check result
arpu

np.float64(284.36)

✔️ An ARPU of 284.36 means the company earned 284.36 in revenue per customer on average in March.

## **2.3. Cost of Goods Sold (COGS)**

**Cost of Goods Sold (COGS)**: represents the direct costs of producing the goods or services a company sells in a specific period. This includes expenses like raw materials, direct labor, and manufacturing overhead.
- **COGS = [Beginning inventory] + [Purchases during the period] - [Ending inventory]**

Since there’s no physical inventory in SaaS businesses, COGS is based on direct delivery costs like software and production, in this case:
- **COGS = [Software Expenses] + [Production Salary]**




### 2.3.1. Software Expenses

#### a. Production Software

In [22]:
# Define production software
production_software = ['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira']

# Total production software expenses in March
software_expenses_202303 = expense_202303[expense_202303['item'].isin(production_software)]['amount'].sum()

# Output result
software_expenses_202303


np.int64(14200)

#### b. Shared Software

Since Slack and Zoom are shared tools used across teams, we allocate 60% of their cost to production, based on the assumption that the majority of usage supports operational delivery.



In [23]:
# Define shared software tools (used across departments)
shared_software = ['Slack', 'Zoom']

# Assump 60% used for production
shared_expense_202303 = expense_202303[expense_202303['item'].isin(shared_software)]['amount'].sum() * 0.6

# Output result
shared_expense_202303


np.float64(864.0)

### 2.3.2. Production salary

In [24]:
# Total payroll of 'Engineering' in March
production_salary_202303 = payroll_202303[payroll_202303['department'] == 'Engineering'] ['paid'].sum()

# Output result
production_salary_202303

np.int64(5200)

### 2.3.3. COGS Calculation

In [25]:
# Calculate COGS
cogs = software_expenses_202303 + shared_expense_202303 + production_salary_202303

# Check result
cogs

np.float64(20264.0)

✔️ The COGS of 20264 represents the total direct costs incurred to deliver service in March.

##**2.4. Gross Margin**

**Gross Margin:** The difference between sales revenue and the cost of goods sold (COGS), expressed as a percentage of sales revenue
- **Gross Margin = (([Revenue] - [COGS]) / [Revenue]) * [100]**




In [26]:
# Calculate gross margin
gross_margin = round(((total_revenue_202303 - cogs) / total_revenue_202303) * 100, 2)

# Check result
gross_margin

np.float64(75.6)

✔️ A gross margin of 75.6% means that for every dollar of revenue, the company retains 75.6 cents after covering the cost of production.

 👉 This is considered high, indicating strong operational efficiency and healthy unit economics.

##**2.5. Customer Lifetime Value (LTV)**

**Customer Lifetime Value (LTV):** A metric that estimates the total revenue a business can expect from a customer over the entire duration of their relationship.

- **LTV = [ARPU] * [Customer Lifespan] * [Gross Margin]**

### 2.5.1. Customer Lifespan

Check 'customer_lifespan_data':

In [27]:
lifespan.sample(5)

Unnamed: 0 start_date churn_date
5         1005 2021-12-06 2022-12-29
10        1010 2021-03-24 2023-03-30
98        1098 2020-06-17 2022-06-13
71        1071 2020-10-01 2022-12-27
75        1075 2022-03-23 2023-03-27

In [28]:
# Calculate difference between churn and start dates
lifespan['lifespan_days'] = pd.to_datetime(lifespan['churn_date']) - pd.to_datetime(lifespan['start_date'])

# Convert timedelta to integer number of days
lifespan['lifespan_days'] = lifespan['lifespan_days'].dt.days

# Show a random sample
lifespan.sample(5)

Unnamed: 0 start_date churn_date  lifespan_days
23        1023 2023-02-17 2023-02-24              7
41        1041 2021-01-05 2023-03-31            815
28        1028 2022-02-13 2022-03-27             42
26        1026 2022-06-27 2022-12-05            161
76        1076 2020-01-25 2021-06-19            511

In [29]:
# Calculate average customer lifespan in months (divide by 30 to convert days to months)
avg_lifespan_month = lifespan['lifespan_days'].mean() / 30

# Check result
avg_lifespan_month

np.float64(9.841333333333333)

### 2.5.2. LTV Calculation

In [30]:
# Calculate LTV
ltv = round(arpu * avg_lifespan_month * (gross_margin/100), 2)

# Check result
ltv

np.float64(2115.65)

✔️ The LTV of 2115.65 means that each customer generates approximately 2,115.65 in total revenue over their average lifespan of 9.8 months.

##**2.6. LTV/CAC**

**LTV/CAC:** A ratio that compares the Customer Lifetime Value (LTV) to the Customer Acquisition Cost (CAC).
This metric is used to assess the efficiency and profitability of a company's customer acquisition strategies.

In [31]:
# Calculate LTV/CAC
ratio_ltv_cac = round(ltv / cac, 2)

# Check result
ratio_ltv_cac

np.float64(1.74)

✔️ An LTV/CAC ratio of 1.74 indicates healthy growth, as the company earns 1.74 for every 1 spent acquiring a customer — well above the break-even point.

#**3. Conclusion and Recommendation**

## **3.1. Conclusion**

Based on the unit economics calculated for Streamline Pro – March 2023:

- The company reported COGS of 20264, resulting in a gross margin of 75.6%, which reflects efficient cost management and strong operational performance.

- The Customer Acquisition Cost (CAC) was 1213.97, while the Customer Lifetime Value (LTV) reached 2115.65.
  
  👉 This yields an LTV/CAC ratio of 1.74, suggesting that the company earns significantly more from each customer than it spends to acquire them.

- With an average customer lifespan of 9.8 months and ARPU of 284.36, Streamline Pro demonstrates profitable customer relationships.

📌 Overall, Streamline Pro is currently in a strong financial position, with healthy margins and efficient customer acquisition, providing a solid foundation for sustainable growth.


## **3.2. Recommendation**

- **Continue monitoring LTV/CAC** monthly to track acquisition performance and identify early signs of inefficiency.

- **Explore strategies to lower CAC**, such as optimizing ad channels or improving conversion rates.

- **Increase LTV** by enhancing retention efforts, upselling features, or raising ARPU through value-added services.

- **Maintain or improve gross margin** by keeping COGS in check, especially as the business scales.